pandas/core/resample.py

```diff
@@ -5,12 +5,13 @@
 from textwrap import dedent
 
 import pandas as pd
-from pandas.core.base import GroupByMixin
-
+from pandas.core.groupby.base import GroupByMixin
+from pandas.core.groupby.ops import BinGrouper, Grouper
 from pandas.core.groupby.groupby import (
-    BinGrouper, Grouper, _GroupBy, GroupBy, SeriesGroupBy, groupby,
-    PanelGroupBy, _pipe_template
+    _GroupBy, GroupBy, groupby, _pipe_template
 )
+from pandas.core.groupby.series import SeriesGroupBy
+from pandas.core.groupby.panel import PanelGroupBy
```

Comment:
Same here, I would import only from `pandas.core.groupby`, and expose there everything that is needed elsewhere in pandas. That way we don't always have to change this if the groupby module is internally refactored.

Reply:
And doing that makes it also clear what is used elsewhere in pandas, as this is what is in the `groupby/__init__.py`

Reply:
this not following the general pattern. It is trivial to update this if imports are changed and IMHO it makes the structure much more clear here.